frontend/src/components/modules/OngoingTournament/PlayoffTournamentView.tsx

```diff
@@ -0,0 +1,137 @@
+
+import "./style.css";
+
+interface TournamentBracketProps {
+  players: string[];
+  initialWinners: string[];
+}
+
+interface Match {
+  players: string[];
+  winners: string[];
+}
+
+const TournamentBracket: React.FC<TournamentBracketProps> = ({
+  players,
+  initialWinners
+}) => {
+  const [rounds, setRounds] = useState<Match[][]>([]);
+  const [winners, setWinners] = useState<string[]>(initialWinners);
+  const [roundNumber, setRoundNumber] = useState<number>(1);
+  const [isRoundFinished, setIsRoundFinished] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (isRoundFinished) {
+      const roundPlayers = roundNumber === 1 ? players : winners;
+      const newRound: Match[] = generateMatches(roundPlayers);
+      setRounds((prevRounds) => [...prevRounds, newRound]);
+      setWinners(newRound.flatMap((match) => match.winners));
+      setRoundNumber((prevRoundNumber) => prevRoundNumber + 1);
+      setIsRoundFinished(false);
+    }
+  }, [isRoundFinished, winners, players, roundNumber]);
+
+  const generateMatches = (roundPlayers: string[]): Match[] => {
+    const matches: Match[] = [];
+
+    for (let i = 0; i < roundPlayers.length; i += 2) {
+      const matchWinner = roundPlayers[i]; // Winner is the left player for simplicity
+
+      const match: Match = {
+        players: [roundPlayers[i], roundPlayers[i + 1]],
+        winners: [matchWinner]
+      };
+
+      matches.push(match);
+    }
+
+    return matches;
+  };
+
+  const handleNextRound = (): void => {
+    setIsRoundFinished(true);
+  };
+
+  return (
+    <Paper className="tournamentBracketRoot" elevation={3}>
+      <Typography variant="h5" gutterBottom>
+        Tournament name
+      </Typography>
+      {rounds.map((round, roundIndex) => (
+        <div key={roundIndex}>
+          <Typography variant="subtitle1">Round {roundIndex + 1}</Typography>
+          <Grid container spacing={2} className="bracketContainer">
+            {round.map((match, matchIndex) => (
+              <Grid
+                key={matchIndex}
+                item
+                xs={6}
+                sm={4}
+                md={3}
+                lg={2}
+                className="matchGridItem"
+              >
+                <div className="matchContainer">
+                  <Button>
+                    {match.players[0]} vs {match.players[1]}
+                    <br></br>
+                    Winner: {match.winners[0]}{" "}
+                    {/* We can print the winner like this or bold the name etc. */}
+                  </Button>
+                </div>
+              </Grid>
+            ))}
+          </Grid>
+        </div>
+      ))}
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={handleNextRound}
+        disabled={isRoundFinished}
+      >
+        Generate Next Round
+      </Button>
+    </Paper>
+  );
+};
+
+const PlayoffTournamentView: React.FC = () => {
```

Comment:
I would probably separate the `PlayOffTournament` and `TournamentBracket` components into their own files.